packages/app/src/systems/Account/components/AccountsDialog/AccountsDialog.tsx

```diff
@@ -0,0 +1,69 @@
+import { cssObj } from '@fuel-ui/css';
+import { Dialog } from '@fuel-ui/react';
+
+import { useAccounts } from '../../hooks';
+import { AddAccount, Logout } from '../../pages';
+import { Accounts } from '../../pages/Accounts';
+
+import { UnlockContent } from '~/systems/Core';
+import { useOverlay } from '~/systems/Overlay';
+
+export function AccountsDialog() {
+  const overlay = useOverlay();
+  const { status, handlers, ...ctx } = useAccounts();
+  const isUnlocking = status('unlocking') || status('unlockingLoading');
+
+  return (
+    <Dialog isOpen={overlay.is((val) => val.includes('accounts'))}>
+      <Dialog.Content css={styles.content}>
+        {isUnlocking && (
+          <UnlockContent
+            unlockText="Add Account"
+            unlockError={ctx.unlockError}
+            onUnlock={handlers.unlock}
+            isLoading={status('unlockingLoading')}
+            onClose={overlay.close}
+          />
+        )}
+        {!isUnlocking && overlay.is('accounts.list') && <Accounts />}
+        {!isUnlocking && overlay.is('accounts.add') && <AddAccount />}
+        {!isUnlocking && overlay.is('accounts.logout') && <Logout />}
```

Comment:
Can we move these checks to the machine?
Also, is it possible to use the enum `Overlays` here?

Reply:
The `overlay.is` is precise to avoid creating a bunch of variables for checking, like I'm doing with `status()` in some hooks. About the enum, it has the `keyof typeof OverlayEnum`, so it's strong typed and will auto-complete. This avoids need to import the enum each time I need it and has the same functionality